environment/src/main/java/com/sequenceiq/environment/environment/validation/network/gcp/GcpEnvironmentNetworkValidator.java

```diff
@@ -49,6 +63,7 @@ public void validateDuringFlow(EnvironmentValidationDto environmentValidationDto
         EnvironmentDto environmentDto = environmentValidationDto.getEnvironmentDto();
         checkSubnetsProvidedWhenExistingNetwork(resultBuilder, networkDto, networkDto.getGcp(),
                 cloudNetworkService.retrieveSubnetMetadata(environmentDto, networkDto));
+        checkAvailabilityZones(resultBuilder, environmentDto, networkDto);
```

Comment:
shouldn't we have it as part of request validation so that environment creation fails. If we have it during flow , we will need to delete environment since we cannot modify zones. If we are caching the region to zones mapping then it should not have any impact on time taken for environment creation.